### ISSUE TITLE
AllowRepeatedCharacters does not work with separators in between

### ISSUE THREAD

**Issue by redzic** [Anders429/word_filter#6]
Hi, if you set `RepeatedCharacterMatchMode::AllowRepeatedCharacters` it does not seem to catch repeated characters with separators in between.
For example:
```
WordFilter::new(
    &[
        "test",
    ],
    &[],
    &[
        ",", "_", ".", ",", "-", "+", "&", "[", "]", ":", ";", "!", "@", "#", "$", "%",
        "*", "(", ")", "<", ">",
    ],
    &[],
    Options {
        repeated_character_match_mode: RepeatedCharacterMatchMode::AllowRepeatedCharacters,
        ..Default::default()
    },
)
```
does not match `t e e s t`,  but it does match `teest`, and `t,_e@s%t`, which seems a little inconsistent.

Thanks for this project btw!

**Comment by Anders429** [comment]
Hey, thanks for the issue! You're right, this does seem to be a bug that I missed in testing.

Your example doesn't actually include a `" "` in the separators list, but even including it still results in the same issue. This doesn't pass that I would expect it to:

```rust
#[test]
fn issue() {
    let filter = WordFilter::new(
        &[
            "test",
        ],
        &[],
        &[
            " ",
        ],
        &[],
    Options {
        repeated_character_match_mode: RepeatedCharacterMatchMode::AllowRepeatedCharacters,
        ..Default::default()
    }
    );

    assert!(filter.check("t e e s t"));
}
```

The issue seems to be that when traversal handles separators, it only pushes the pointer's `current_node` to the `return_nodes` `Vec`, when it should also branch another `Pointer` that also pushes the previous pointer's `current_node` to the `return_nodes` `Vec`. 

Looks like a pretty simple fix. Thanks again for finding it :)

**Comment by Anders429** [comment]
Should be fixed now. I have published the change as version `0.2.0` on [crates.io](https://crates.io/crates/word_filter).

Feel free to reopen if this doesn't resolve the issue. Thanks again!
